src/python/sortingview/gui/WorkspacePage/WorkspaceNavigationComponent/WorkspaceNavigationComponent.tsx

```diff
@@ -37,13 +37,13 @@ const WorkspacePart: FunctionComponent<Props> = ({workspaceRoute, workspaceRoute
 
 const RecordingPart: FunctionComponent<Props> = ({workspace, workspaceRoute, workspaceRouteDispatch}) => {
     const handleClick = useCallback(() => {
-        if ((workspaceRoute.page !== 'recording') && (workspaceRoute.page !== 'sorting')) throw Error('Unexpected')
+        if ((workspaceRoute.page !== 'recording') && (workspaceRoute.page !== 'sorting') && (workspaceRoute.page !== 'sortingComparison')) throw Error('Unexpected')
```

Comment:
Is there a meaningful name we could apply to this grouping of pages? Like are the recording, sorting, and sortingComparison pages the complete list of known pages? I see the list appearing both here and a few lines below, and it seems like they could get out of sync & it might be clearer what's happening if we could say something like `! pageRoutesDefiningRecordingId.includes(workspaceRoute,page)`.

Reply:
This particular line is serving as a type guard for typescript. If typescript knows it is either a recording or sorting page, then it knows that it will have a recordingId. It wouldn't work the same if we had a list... like `['recording', 'sorting'].includes(...)` because typescript wouldn't recognize the implications. So it sort of has to be the way it is ... until we do something more sophisticated with the workspaceRoute type.